File: outpost_assessment/outpost_assessment/page/attendance_dashboard/attendance_dashboard.py
# Copyright (c) 2026, Dharanipathi and contributors
# For license information, please see license.txt

import os
import json
import frappe

@frappe.whitelist()
def get_attendance_data():
	app_path = frappe.get_app_path("outpost_assessment")
	mock_data_path = os.path.join(app_path, "outpost_assessment", "page", "attendance_dashboard", "mock_data.json")
	
	if os.path.exists(mock_data_path):
		try:
			with open(mock_data_path, "r", encoding="utf-8") as f:
				return json.load(f)
		except Exception as e:
			frappe.log_error(message=str(e), title="Error loading mock_data.json")
			return []
			
	return []

File: outpost_assessment/outpost_assessment/page/attendance_dashboard/attendance_dashboard.js
// Copyright(c) 2026, Dharanipathi and contributors
// For license information, please see license.txt

frappe.pages["attendance-dashboard"].on_page_load = function (wrapper) {
	frappe.ui.make_app_page({
		parent: wrapper,
		title: __("Attendance Dashboard"),
		single_column: true,
	});
};

frappe.pages["attendance-dashboard"].on_page_show = function (wrapper) {
	load_attendance_dashboard(wrapper);
};

function load_attendance_dashboard(wrapper) {
	const $parent = $(wrapper).find(".layout-main-section");
	$parent.empty();

	frappe.require("attendance_dashboard.bundle.js").then(() => {
		frappe.attendance_dashboard = new frappe.ui.AttendanceDashboard({
			wrapper: $parent,
			page: wrapper.page,
		});
	});
}
